packages/tiny-router/src/actions.ts

```diff
@@ -44,8 +44,28 @@ export const set: TinyRouter["actions"]["router"]["set"] = ({
   if (libraries.source && libraries.source.normalize)
     link = libraries.source.normalize(link);
 
-  // Set state default value.
-  if (!options.state) options.state = {};
+  // Clone the state that we are going to use for `window.history` because it
+  // cannot contain proxies.
+  options.state = JSON.parse(JSON.stringify(options.state || {}));
+
+  // If the data is a redirection, then we set the link to the location.
+  // The redirections are stored in source.data just like any other data.
+  const data = state.source?.get(link);
+  if (data && isRedirection(data)) {
+    if (data.isExternal) {
+      window.replaceLocation(data.location);
+    } else {
+      // If the link is internal, we have to discard the domain.
+      const { pathname, hash, search } = new URL(
+        data.location,
+        "https://dummy-domain.com"
```

Comment:
Yeah, I thought so at the beginning because the standard (https://tools.ietf.org/html/rfc2616#section-14.30) says that `Location` should contain absolute links, but people can still use relative links and they work fine in the browsers, so we need to make sure they work.

_Most of the redirections we used in WordPress for the e2e tests are actually relative links._